From db0a845c99a6fb88a6bd52ea3c0f58306f82b515 Mon Sep 17 00:00:00 2001
From: Demon000 <demonsingur@gmail.com>
Date: Thu, 11 Jan 2018 08:17:57 +0200
Subject: [PATCH 07/20] legacy: add back rules for non-treble devices

All these rules have been removed as part of commit
c7def121c0a13733060290f9ce1401ca7cf44405, but most of them are
needed for old blobs.

Stuff left out:
 * imscm, it must be re-done since imscm is gone completely
 * init_shell, the shell files it tries to execute can be moved to vendor
   and fixed to use vendor shell
 * dataservice_app, the changes get uncommented later
 * qti-logkit, shouldn't be present on devices
 * dpmd related rules, because dpmd has been moved to private
   and it can be just removed in pretty much all SOCs

Unused stuff left out:
 * atfwd
 * audiod
 * dtseagleservice
 * esepmdaemon
 * fidodaemon
 * hal_fps
 * hbtp
 * logdumpd
 * mdtp
 * mmi
 * qfp-daemon
 * qsee_svc_app
 * RIDL
 * secotad
 * seempd
 * spdaemon
 * usf
 * wfdservice

Change-Id: I36ea1826bc042591f10859df35cf5924ac663088
---
 legacy-common/app.te                   |  9 +++++++++
 legacy-common/audioserver.te           |  9 +++++++++
 legacy-common/bluetooth.te             |  3 +++
 legacy-common/cameraserver.te          |  5 +++++
 legacy-common/cnd.te                   | 13 +++++++++++++
 legacy-common/dhcp.te                  |  2 ++
 legacy-common/hal_audio.te             |  7 +++++++
 legacy-common/hal_camera.te            |  7 +++++++
 legacy-common/hal_graphics_composer.te | 10 ++++++++++
 legacy-common/ims.te                   |  5 +++++
 legacy-common/iop.te                   |  1 +
 legacy-common/irsc_util.te             |  1 +
 legacy-common/location.te              | 10 ++++++++++
 legacy-common/location_app.te          |  2 ++
 legacy-common/mdm_helper.te            |  1 +
 legacy-common/mediacodec.te            |  3 +++
 legacy-common/mediaserver.te           | 14 ++++++++++++++
 legacy-common/mm-pp-daemon.te          | 12 ++++++++++++
 legacy-common/mm-qcamerad.te           |  7 +++++++
 legacy-common/net.te                   |  1 +
 legacy-common/netd.te                  |  4 ++++
 legacy-common/netmgrd.te               |  4 ++++
 legacy-common/peripheral_manager.te    |  2 ++
 legacy-common/qlogd.te                 |  1 +
 legacy-common/qseecomd.te              |  4 ++++
 legacy-common/qseeproxy.te             |  4 ++++
 legacy-common/qti.te                   |  1 +
 legacy-common/rild.te                  |  2 ++
 legacy-common/sensors.te               |  2 ++
 legacy-common/surfaceflinger.te        |  6 ++++++
 legacy-common/system_app.te            | 23 +++++++++++++++++++++++
 legacy-common/system_server.te         | 24 ++++++++++++++++++++++++
 legacy-common/untrusted_app.te         |  2 ++
 legacy-common/wcnss_service.te         |  1 +
 34 files changed, 202 insertions(+)
 create mode 100644 legacy-common/app.te
 create mode 100644 legacy-common/audioserver.te
 create mode 100644 legacy-common/bluetooth.te
 create mode 100644 legacy-common/cameraserver.te
 create mode 100644 legacy-common/cnd.te
 create mode 100644 legacy-common/dhcp.te
 create mode 100644 legacy-common/hal_audio.te
 create mode 100644 legacy-common/hal_camera.te
 create mode 100644 legacy-common/hal_graphics_composer.te
 create mode 100644 legacy-common/ims.te
 create mode 100644 legacy-common/iop.te
 create mode 100644 legacy-common/irsc_util.te
 create mode 100644 legacy-common/location.te
 create mode 100644 legacy-common/location_app.te
 create mode 100644 legacy-common/mdm_helper.te
 create mode 100644 legacy-common/mediacodec.te
 create mode 100644 legacy-common/mediaserver.te
 create mode 100644 legacy-common/mm-pp-daemon.te
 create mode 100644 legacy-common/mm-qcamerad.te
 create mode 100644 legacy-common/net.te
 create mode 100644 legacy-common/netd.te
 create mode 100644 legacy-common/netmgrd.te
 create mode 100644 legacy-common/peripheral_manager.te
 create mode 100644 legacy-common/qlogd.te
 create mode 100644 legacy-common/qseecomd.te
 create mode 100644 legacy-common/qseeproxy.te
 create mode 100644 legacy-common/qti.te
 create mode 100644 legacy-common/rild.te
 create mode 100644 legacy-common/sensors.te
 create mode 100644 legacy-common/surfaceflinger.te
 create mode 100644 legacy-common/system_app.te
 create mode 100644 legacy-common/system_server.te
 create mode 100644 legacy-common/untrusted_app.te
 create mode 100644 legacy-common/wcnss_service.te

diff --git a/legacy-common/app.te b/legacy-common/app.te
new file mode 100644
index 0000000..e4f1ff4
--- /dev/null
+++ b/legacy-common/app.te
@@ -0,0 +1,9 @@
+unix_socket_connect(appdomain, cnd, cnd)
+unix_socket_connect(appdomain, qlogd, qlogd)
+
+allow appdomain mpctl_socket:dir r_dir_perms;
+
+unix_socket_send(appdomain, mpctl, perfd)
+unix_socket_connect(appdomain, mpctl, perfd)
+unix_socket_send(appdomain, mpctl, mpdecision)
+unix_socket_connect(appdomain, mpctl, mpdecision)
diff --git a/legacy-common/audioserver.te b/legacy-common/audioserver.te
new file mode 100644
index 0000000..91d73e0
--- /dev/null
+++ b/legacy-common/audioserver.te
@@ -0,0 +1,9 @@
+# Allow socket communication with perfd
+allow audioserver mpctl_socket:dir r_dir_perms;
+unix_socket_send(audioserver, mpctl, mpdecision)
+unix_socket_connect(audioserver, mpctl, mpdecision)
+unix_socket_send(audioserver, mpctl, perfd)
+unix_socket_connect(audioserver, mpctl, perfd)
+
+# Allow communication with peripheral manager
+use_per_mgr(audioserver)
diff --git a/legacy-common/bluetooth.te b/legacy-common/bluetooth.te
new file mode 100644
index 0000000..927c368
--- /dev/null
+++ b/legacy-common/bluetooth.te
@@ -0,0 +1,3 @@
+qmux_socket(bluetooth)
+
+allow bluetooth wcnss_filter:unix_stream_socket connectto;
diff --git a/legacy-common/cameraserver.te b/legacy-common/cameraserver.te
new file mode 100644
index 0000000..8eb8d59
--- /dev/null
+++ b/legacy-common/cameraserver.te
@@ -0,0 +1,5 @@
+allow cameraserver mm-qcamerad:unix_dgram_socket sendto;
+
+unix_socket_connect(cameraserver, mpctl, perfd)
+unix_socket_connect(cameraserver, thermal, thermal-engine)
+unix_socket_connect(cameraserver, sensors, sensors)
diff --git a/legacy-common/cnd.te b/legacy-common/cnd.te
new file mode 100644
index 0000000..7f7263e
--- /dev/null
+++ b/legacy-common/cnd.te
@@ -0,0 +1,13 @@
+# Allow communication with udp sockets created by apps
+allow cnd appdomain:udp_socket rw_socket_perms;
+
+# Allow hostapd_cli launch
+allow cnd shell_exec:file rx_file_perms;
+
+# Allow communication with inet sockets
+cnd_nims_socket_perm(appdomain)
+cnd_nims_socket_perm(system_server)
+cnd_nims_socket_perm(mediaserver)
+cnd_nims_socket_perm(mtp)
+cnd_nims_socket_perm(wfdservice)
+cnd_nims_socket_perm(drmserver)
diff --git a/legacy-common/dhcp.te b/legacy-common/dhcp.te
new file mode 100644
index 0000000..e2ad60c
--- /dev/null
+++ b/legacy-common/dhcp.te
@@ -0,0 +1,2 @@
+# Allow socket communication with cnd
+unix_socket_connect(dhcp, cnd, cnd)
diff --git a/legacy-common/hal_audio.te b/legacy-common/hal_audio.te
new file mode 100644
index 0000000..1e0b10a
--- /dev/null
+++ b/legacy-common/hal_audio.te
@@ -0,0 +1,7 @@
+# Allow socket communication with perfd
+unix_socket_send(hal_audio, mpctl, perfd)
+unix_socket_connect(hal_audio, mpctl, perfd)
+
+# Allow socket creation for audio arbitration
+allow hal_audio audio_data_file:sock_file { create setattr unlink };
+allow hal_audio audio_data_file:dir remove_name;
diff --git a/legacy-common/hal_camera.te b/legacy-common/hal_camera.te
new file mode 100644
index 0000000..9beafbc
--- /dev/null
+++ b/legacy-common/hal_camera.te
@@ -0,0 +1,7 @@
+# Allow binder communication
+binder_use(hal_camera)
+allow hal_camera surfaceflinger_service:service_manager find;
+
+# Allow socket communication with perfd
+unix_socket_connect(hal_camera, mpctl, perfd)
+unix_socket_send(hal_camera, mpctl, perfd)
diff --git a/legacy-common/hal_graphics_composer.te b/legacy-common/hal_graphics_composer.te
new file mode 100644
index 0000000..1e69f00
--- /dev/null
+++ b/legacy-common/hal_graphics_composer.te
@@ -0,0 +1,10 @@
+# Allow binder communication
+binder_use(hal_graphics_composer)
+binder_service(hal_graphics_composer_default)
+allow hal_graphics_composer power_service:service_manager find;
+
+# Allow socket communication with perfd
+unix_socket_connect(hal_graphics_composer, mpctl, perfd)
+
+# Allow socket communication with post process daemon
+unix_socket_connect(hal_graphics_composer, pps, mm-pp-daemon)
diff --git a/legacy-common/ims.te b/legacy-common/ims.te
new file mode 100644
index 0000000..facd4b7
--- /dev/null
+++ b/legacy-common/ims.te
@@ -0,0 +1,5 @@
+# Allow binder communication
+binder_use(ims)
+
+# Allow NDC command run
+allow ims shell_exec:file rx_file_perms;
diff --git a/legacy-common/iop.te b/legacy-common/iop.te
new file mode 100644
index 0000000..72dd0a8
--- /dev/null
+++ b/legacy-common/iop.te
@@ -0,0 +1 @@
+allow dumpstate iop_socket:sock_file rw_file_perms;
diff --git a/legacy-common/irsc_util.te b/legacy-common/irsc_util.te
new file mode 100644
index 0000000..a029e67
--- /dev/null
+++ b/legacy-common/irsc_util.te
@@ -0,0 +1 @@
+domain_auto_trans(shell, irsc_util_exec, irsc_util)
diff --git a/legacy-common/location.te b/legacy-common/location.te
new file mode 100644
index 0000000..8f2228e
--- /dev/null
+++ b/legacy-common/location.te
@@ -0,0 +1,10 @@
+# Allow binder communication
+binder_use(location)
+allow location { sensorservice_service permission_service }:service_manager find;
+
+# Allow sh usage
+allow location shell_exec:file rx_file_perms;
+
+# Allow socket communication
+allow location wifi_data_file:sock_file create_file_perms;
+allow location system_server:unix_stream_socket { read write connectto};
diff --git a/legacy-common/location_app.te b/legacy-common/location_app.te
new file mode 100644
index 0000000..28c744c
--- /dev/null
+++ b/legacy-common/location_app.te
@@ -0,0 +1,2 @@
+# Allow binder communication
+allow location_app { app_api_service system_api_service }:service_manager find;
diff --git a/legacy-common/mdm_helper.te b/legacy-common/mdm_helper.te
new file mode 100644
index 0000000..7d1f14a
--- /dev/null
+++ b/legacy-common/mdm_helper.te
@@ -0,0 +1 @@
+allow mdm_helper shell_exec:file rx_file_perms;
diff --git a/legacy-common/mediacodec.te b/legacy-common/mediacodec.te
new file mode 100644
index 0000000..399d77b
--- /dev/null
+++ b/legacy-common/mediacodec.te
@@ -0,0 +1,3 @@
+# Allow binder communication
+allow mediacodec wfdservice_service:service_manager find;
+allow mediacodec audioserver_service:service_manager find;
diff --git a/legacy-common/mediaserver.te b/legacy-common/mediaserver.te
new file mode 100644
index 0000000..2ce3fc8
--- /dev/null
+++ b/legacy-common/mediaserver.te
@@ -0,0 +1,14 @@
+qmux_socket(mediaserver)
+
+unix_socket_connect(mediaserver, cnd, cnd)
+unix_socket_send(mediaserver, camera, mm-qcamerad)
+
+allow mediaserver mpctl_socket:dir r_dir_perms;
+unix_socket_send(mediaserver, mpctl, perfd)
+unix_socket_connect(mediaserver, mpctl, perfd)
+unix_socket_send(mediaserver, mpctl, mpdecision)
+unix_socket_connect(mediaserver, mpctl, mpdecision)
+
+unix_socket_connect(mediaserver, thermal, thermal-engine)
+
+allow mediaserver time_daemon:unix_stream_socket connectto;
diff --git a/legacy-common/mm-pp-daemon.te b/legacy-common/mm-pp-daemon.te
new file mode 100644
index 0000000..8385911
--- /dev/null
+++ b/legacy-common/mm-pp-daemon.te
@@ -0,0 +1,12 @@
+# Allow binder communication
+binder_use(mm-pp-daemon)
+allow mm-pp-daemon {
+	surfaceflinger_service
+	sensorservice_service
+	permission_service
+	power_service
+}:service_manager find;
+
+allow mm-pp-daemon { shell_exec zygote_exec }:file rx_file_perms;
+
+allow mm-pp-daemon system_server:unix_stream_socket rw_socket_perms;
diff --git a/legacy-common/mm-qcamerad.te b/legacy-common/mm-qcamerad.te
new file mode 100644
index 0000000..87ce39a
--- /dev/null
+++ b/legacy-common/mm-qcamerad.te
@@ -0,0 +1,7 @@
+binder_use(mm-qcamerad)
+
+allow mm-qcamerad camera_socket:sock_file { create unlink write };
+allow mm-qcamerad camera_data_file:sock_file { create unlink };
+
+allow mm-qcamerad system_server:unix_stream_socket rw_socket_perms;
+allow mm-qcamerad sensorservice_service:service_manager find;
diff --git a/legacy-common/net.te b/legacy-common/net.te
new file mode 100644
index 0000000..66582e5
--- /dev/null
+++ b/legacy-common/net.te
@@ -0,0 +1 @@
+unix_socket_connect(netdomain, cnd, cnd)
diff --git a/legacy-common/netd.te b/legacy-common/netd.te
new file mode 100644
index 0000000..c550eed
--- /dev/null
+++ b/legacy-common/netd.te
@@ -0,0 +1,4 @@
+unix_socket_connect(netd, cnd, cnd)
+allow netd wfdservice:tcp_socket rw_socket_perms;
+allow netd hostapd:unix_dgram_socket sendto;
+allow netd wpa_socket:sock_file create_file_perms;
diff --git a/legacy-common/netmgrd.te b/legacy-common/netmgrd.te
new file mode 100644
index 0000000..f449e9c
--- /dev/null
+++ b/legacy-common/netmgrd.te
@@ -0,0 +1,4 @@
+allow netmgrd { wcnss_service_exec shell_exec toolbox_exec }:file rx_file_perms;
+
+allow netmgrd netd_socket:sock_file w_file_perms;
+r_dir_file(netmgrd, net_data_file)
diff --git a/legacy-common/peripheral_manager.te b/legacy-common/peripheral_manager.te
new file mode 100644
index 0000000..493adfc
--- /dev/null
+++ b/legacy-common/peripheral_manager.te
@@ -0,0 +1,2 @@
+binder_use(per_mgr)
+binder_service(per_mgr)
diff --git a/legacy-common/qlogd.te b/legacy-common/qlogd.te
new file mode 100644
index 0000000..5d81951
--- /dev/null
+++ b/legacy-common/qlogd.te
@@ -0,0 +1 @@
+allow qlogd shell_exec:file rx_file_perms;
diff --git a/legacy-common/qseecomd.te b/legacy-common/qseecomd.te
new file mode 100644
index 0000000..f2cf3b0
--- /dev/null
+++ b/legacy-common/qseecomd.te
@@ -0,0 +1,4 @@
+binder_use(tee)
+allow tee surfaceflinger_service:service_manager  find;
+
+allow tee system_app:unix_dgram_socket sendto;
diff --git a/legacy-common/qseeproxy.te b/legacy-common/qseeproxy.te
new file mode 100644
index 0000000..ec6ae92
--- /dev/null
+++ b/legacy-common/qseeproxy.te
@@ -0,0 +1,4 @@
+binder_use(qseeproxy)
+binder_service(qseeproxy)
+
+allow qseeproxy system_app:unix_dgram_socket sendto;
diff --git a/legacy-common/qti.te b/legacy-common/qti.te
new file mode 100644
index 0000000..f9001f8
--- /dev/null
+++ b/legacy-common/qti.te
@@ -0,0 +1 @@
+allow qti shell_exec:file rx_file_perms;
diff --git a/legacy-common/rild.te b/legacy-common/rild.te
new file mode 100644
index 0000000..ef86beb
--- /dev/null
+++ b/legacy-common/rild.te
@@ -0,0 +1,2 @@
+binder_use(rild)
+allow rild { mediaserver_service audioserver_service }:service_manager find;
diff --git a/legacy-common/sensors.te b/legacy-common/sensors.te
new file mode 100644
index 0000000..d6c701a
--- /dev/null
+++ b/legacy-common/sensors.te
@@ -0,0 +1,2 @@
+binder_use(sensors)
+binder_call(sensors, servicemanager)
diff --git a/legacy-common/surfaceflinger.te b/legacy-common/surfaceflinger.te
new file mode 100644
index 0000000..4522d88
--- /dev/null
+++ b/legacy-common/surfaceflinger.te
@@ -0,0 +1,6 @@
+allow surfaceflinger mpctl_socket:dir r_dir_perms;
+unix_socket_send(surfaceflinger, mpctl, perfd)
+unix_socket_connect(surfaceflinger, mpctl, perfd)
+unix_socket_send(surfaceflinger, mpctl, mpdecision)
+unix_socket_connect(surfaceflinger, mpctl, mpdecision)
+unix_socket_connect(surfaceflinger, pps, mm-pp-daemon)
diff --git a/legacy-common/system_app.te b/legacy-common/system_app.te
new file mode 100644
index 0000000..b0016fd
--- /dev/null
+++ b/legacy-common/system_app.te
@@ -0,0 +1,23 @@
+allow system_app mpctl_socket:dir r_dir_perms;
+unix_socket_connect(system_app, mpctl, mpdecision)
+unix_socket_connect(system_app, mpctl, perfd)
+unix_socket_send(system_app, mpctl, mpdecision)
+unix_socket_send(system_app, mpctl, perfd)
+
+allow system_app qseeproxy:unix_dgram_socket sendto;
+allow system_app tee:unix_dgram_socket sendto;
+
+allow system_app wcnss_filter:unix_stream_socket connectto;
+unix_socket_send(system_app, wififtmd, wifi_ftmd)
+
+allow system_app time_daemon:unix_stream_socket connectto;
+
+allow system_app usf:unix_stream_socket connectto;
+
+unix_socket_connect(system_app, ims, ims)
+
+unix_socket_connect(system_app, pps, mm-pp-daemon)
+
+# Allow binder communication with ims
+binder_call(system_app, ims)
+allow system_app ims_service:service_manager find;
diff --git a/legacy-common/system_server.te b/legacy-common/system_server.te
new file mode 100644
index 0000000..11bb383
--- /dev/null
+++ b/legacy-common/system_server.te
@@ -0,0 +1,24 @@
+allow system_server mpctl_socket:dir r_dir_perms;
+unix_socket_connect(system_server, gamed, gamed)
+unix_socket_connect(system_server, mpctl, mpdecision)
+unix_socket_connect(system_server, mpctl, perfd)
+unix_socket_send(system_server, mpctl, mpdecision)
+unix_socket_send(system_server, mpctl, perfd)
+
+netmgr_socket(system_server)
+qmux_socket(system_server)
+use_per_mgr(system_server)
+
+allow system_server sensors:unix_stream_socket sendto;
+allow system_server sensors_socket:sock_file r_file_perms;
+unix_socket_connect(system_server, sensors, sensors)
+unix_socket_send(system_server, sensors, sensors)
+
+allow system_server location:unix_stream_socket connectto;
+allow system_server location_socket:sock_file create_file_perms;
+
+allow system_server usf:unix_stream_socket connectto;
+
+allow system_server wcnss_filter:unix_stream_socket connectto;
+
+unix_socket_connect(system_server, cnd, cnd)
diff --git a/legacy-common/untrusted_app.te b/legacy-common/untrusted_app.te
new file mode 100644
index 0000000..147e006
--- /dev/null
+++ b/legacy-common/untrusted_app.te
@@ -0,0 +1,2 @@
+unix_socket_connect(untrusted_app, mpctl, mpdecision)
+unix_socket_send(untrusted_app, mpctl, mpdecision)
diff --git a/legacy-common/wcnss_service.te b/legacy-common/wcnss_service.te
new file mode 100644
index 0000000..93a8d5b
--- /dev/null
+++ b/legacy-common/wcnss_service.te
@@ -0,0 +1 @@
+binder_use(wcnss_service)
-- 
2.7.4

